From 7491c46af375141eb7a6ffcd2d8c9d4bb0df74c1 Mon Sep 17 00:00:00 2001
From: Tronikel <contactdonatas@gmail.com>
Date: Sat, 9 Nov 2024 17:07:05 +0200
Subject: [PATCH] fix: make sure nvim cmp is prepared

---
 lua/cmp/init.lua | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/lua/cmp/init.lua b/lua/cmp/init.lua
index 6867679c2..2d9cae784 100644
--- a/lua/cmp/init.lua
+++ b/lua/cmp/init.lua
@@ -328,9 +328,13 @@ cmp.setup = setmetatable({
   end,
 })
 
+local cmp_prepared = false
+
 -- In InsertEnter autocmd, vim will detects mode=normal unexpectedly.
 local on_insert_enter = function()
-  if config.enabled() then
+  if not cmp_prepared or config.enabled() then
+    cmp_prepared = true
+
     cmp.config.compare.scopes:update()
     cmp.config.compare.locality:update()
     cmp.core:prepare()
